propagation/propagation.go

```diff
@@ -29,6 +29,15 @@ type TextMapCarrier interface {
 	// must never be done outside of a new major release.
 }
 
+// MultiTextMapCarrier is a TextMapCarrier that can return multiple values for a single key.
```

Comment:
There is no spec proposal or discussion to inject multiple instances of the same header, only to extract multiple from incoming headers (so it can align with W3C spec)

Reply:
How would you propagate the multiple values further when doing outgoing requests?

Reply:
https://github.com/open-telemetry/opentelemetry-specification/issues/433#issuecomment-2479342313